# Visualize / Stack `gctree` visualizations

This notebook will read in the gctree svg's from each PR and stack them into single images for easier comparison.

In [2]:
import os
import sys

import pandas as pd
import svg_stack as ss
from IPython.display import SVG as ip_SVG
from IPython.display import display as ip_display
import svgutils.transform as sg
from svgutils.compose import *

/tmp/ipykernel_14377/1927145887.py:4: DeprecationWarning: 
Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
but was not found to be installed on your system.
If this would cause problems for you,
please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466
        
  import pandas as pd


In [7]:
manifest_path = '../metadata.PR1.csv'
# results_dir = "results/archive/2024-01-19-full/gctrees"
results_dir = "results/gctrees"
svg_filename = "delta_bind.svg"
# svg_filename = "gctree.inference.1.svg"
output_dir = "output/gctree_svg_stack_imm_duration"

In [8]:
if not os.path.exists(output_dir):
    os.makedirs(output_dir)

manifest_df = pd.read_csv(manifest_path)
manifest_df.head()
# manifest_df.query(f"PR == 2.01")

PR imm_duration strain  mouse  plate  gc node cell_type hc_barcode  \
0  1.04          d20     wt      1      2   1   RP        GC          5   
1  1.04          d20     wt      1      3   2   RP        GC          6   
2  1.08          d20     wt      1      4   3   RP        GC          5   
3  1.08          d20     wt      1      5   4   LP        GC          6   
4  1.08          d20     wt      1      6   5   LP        GC          7   

  lc_barcode              row                         col  
0          5  A.B.C.D.E.F.G.H  1.2.3.4.5.6.7.8.9.10.11.12  
1          6  A.B.C.D.E.F.G.H  1.2.3.4.5.6.7.8.9.10.11.12  
2          5  A.B.C.D.E.F.G.H  1.2.3.4.5.6.7.8.9.10.11.12  
3          6  A.B.C.D.E.F.G.H  1.2.3.4.5.6.7.8.9.10.11.12  
4          7  A.B.C.D.E.F.G.H  1.2.3.4.5.6.7.8.9.10.11.12

In [9]:
print(manifest_df.PR.unique())

[1.04 1.08 1.07 1.05 1.03 1.06 1.01 1.02 1.09 1.1 ]


In [10]:

# def create_subpanel(svg_path, annotation, position, scale=0.5):

# loop through each row, and find the corresponding tree in the results directory named "PR<PR>-<mouse>-<node>-<gc>-GC"
for duration, duration_df in manifest_df.groupby('imm_duration'):
    # Create an empty figure
    fig = sg.SVGFigure()

    # Initial y position for stacking
    y_offset = 60
    all_panels = []
    all_panels.append(Text(f"IMM Duration: {duration}", 550, 15, size=16, weight='bold'))

    for pr, pr_df in duration_df.groupby('PR'):

        pr_str = f"{pr:.2f}"

        for i, row in pr_df.iterrows():

            position=(10, y_offset)
            svg_path = f"{results_dir}/PR{pr_str}-{row['mouse']}-{row['node']}-{row['gc']}-GC/{svg_filename}"
            annotation = f"PR: {pr_str}, Mouse: {row['mouse']}, Node: {row['node']}, GC: {row['gc']}"
            # print(svg_path)
            if not os.path.exists(svg_path):
                print(f"Skipping {svg_path}")
                # panel = Panel(
                #   Text(f"(!) No tree found for: {annotation}", 10, 0, size=12, weight='bold', color="red")
                # #   Text(f"(!) No tree found for: {svg_path}", 10, 0, size=12, weight='bold', color="red")
                # ).move(*position)
                # all_panels.append(panel)
            else:
                continue
                # Create a subpanel
                # panel = Panel(
                #         SVG(svg_path).scale(0.3),
                #         Text(annotation, 10, -15, size=12, weight='bold')
                # ).move(*position)
                # all_panels.append(panel)

                # y_offset += 250

    # fig = Figure("%dpx" % 1250, "%dpx" % y_offset, *all_panels)
    # fig.save(f"{output_dir}/{duration}.{svg_filename}")
    # ip_display(ip_SVG(f"{output_dir}/{duration}.{svg_filename}"))

Skipping results/gctrees/PR1.09-11-RP-112-GC/delta_bind.svg
Skipping results/gctrees/PR1.09-11-RB-116-GC/delta_bind.svg


PR: 1.09, Mouse: 11, Node: RP, GC: 112

PR: 1.09, Mouse: 11, Node: RB, GC:116

PR: 2.02, Mouse: 12 Node:LP, GC: 12B

PR: 2.04, Mouse: 1 Node: RP, GC: 1B